jvcl/run/JvJVCLUtils.pas

```diff
@@ -2291,10 +2291,7 @@ procedure ShadeRect(DC: HDC; const Rect: TRect);
 
 function ScreenWorkArea: TRect;
 begin
-  {$IFDEF MSWINDOWS}
-  if not SystemParametersInfo(SPI_GETWORKAREA, 0, @Result, 0) then
-  {$ENDIF MSWINDOWS}
-  Result := Bounds(0, 0, Screen.Width, Screen.Height);
+  Result := Screen.MonitorFromWindow(Screen.ActiveForm.Handle).WorkareaRect;
```

Comment:
This should use `Screen.ActiveCustomForm` to be even more inclusive

Reply:
There is an issue I am having with the FIX provided above in Rio D10.3.2 (Parallels VM, Win 10). It works fine if I use the TJvDesktopAlert directly from a form, ie. I drag the control onto a form and use it from there, **but if I create it in a DLL and use JvDesktopAlert.Execute I get an AV at this line in JvJVCLUtils.pas, function ScreenWorkArea at line 2308:**

`Result := Screen.MonitorFromWindow(Screen.ActiveCustomForm.Handle).WorkareaRect;`
>  Access violation at address 05E9C20F in module 'SFMS_UI.dll'. Read of address 00000288

The result is 100% repeatable and appears to be whenever I try to access the Screen.ActiveCustomForm.Handle although I have confirmed that Screen.ActiveCustomForm.Handle appears to be returning what appears to be a valid value. I get the error even if I try something simple like copying that Handle to a Handle variable:

`myHandle := Screen.ActiveCustomeForm.Handle`

This appeared to work from a DLL with Tokyo D10.2.3 for months. **If I rollback to the original code, it works fine in RIO when called from/executed from a DLL.** Bottom line that Fix does not work for me IF I create and use the JvDesktopAlert from a DLL and it's RIO D10.3.2. If you need my DLL code let me know, but I just create the Alert and use it passing in the title and message. It is getting to all of the relevant places including those in VCL.Forms and JvJVCLUtils. There is little or no documentation on any of these units that I can find.

Reply:
This is most likely because `Screen.ActiveCustomForm` returns `nil`
As you are calling from a DLL, this is totally understandable because the VCL inside the DLL has no knowledge of the host process, and it may not even be made with Delphi.
The solution would be to store `Screen.ActiveCustomForm` in a variable and if that variable is `Assigned` then use it to call `Screen.MonitorFromWindow`. In the `else`part, I would use the original code.

Reply:
Thanks for the response. Yes it is being called from a DLL. If FIX code called from the mainform either creating the component or with the component on the form it works fine. Call the FIX  code from a DLL and the errors described occur. 
I will implement your suggestion.

Reply:
FYI, I just pushed a revision 70d24112c with the suggested fix